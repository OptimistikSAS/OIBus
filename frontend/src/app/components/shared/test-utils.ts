import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
