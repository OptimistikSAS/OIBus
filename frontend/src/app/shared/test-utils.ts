import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
