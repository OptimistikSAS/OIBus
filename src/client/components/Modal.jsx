import React from 'react'
import PropTypes from 'prop-types'
import { Button, Modal as BsModal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap'

const Modal = ({ children, show, title, body, acceptLabel, denyLabel, acceptColor }) => {
  const [open, setOpen] = React.useState(show)
  const [callback, setCallback] = React.useState(null)

  /**
   * Shows the confirmation modal
   * @param {func} callbackParam The function to be called upon confirmation
   * @param {Object} event The event object
   * @returns {void}
   */
  const showModal = (callbackParam) => (event) => {
    event.stopPropagation()
    setOpen(true)
    setCallback({ func: callbackParam })
  }

  /**
   * Hides the modal
   * @returns {void}
   */
  const hideModal = () => {
    setOpen(false)
    setCallback(null)
  }

  /**
   * Handles the confirmation of the modal
   * @returns {void}
   */
  const confirm = () => {
    callback.func()
    hideModal()
  }

  return (
    <>
      {children(showModal)}
      <BsModal isOpen={open} toggle={hideModal}>
        <ModalHeader toggle={hideModal}>{title}</ModalHeader>
        <ModalBody>{body}</ModalBody>
        <ModalFooter>
          <Button color={acceptColor} onClick={confirm}>
            {acceptLabel}
          </Button>
          {' '}
          <Button color="secondary" onClick={hideModal}>
            {denyLabel}
          </Button>
        </ModalFooter>
      </BsModal>
    </>
  )
}
Modal.propTypes = {
  show: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  acceptLabel: PropTypes.string,
  denyLabel: PropTypes.string,
  children: PropTypes.func.isRequired,
  acceptColor: PropTypes.string,
}

Modal.defaultProps = {
  acceptLabel: 'Confirm',
  denyLabel: 'Cancel',
  acceptColor: 'primary',
}
export default Modal
