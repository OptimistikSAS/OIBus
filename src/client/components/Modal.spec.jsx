import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import Modal from './Modal.jsx'

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('Modal', () => {
  test('check Modal', () => {
    act(() => {
      ReactDOM.render(
        <Modal
          show
          title="title"
          body="body content"
          acceptLabel="accept"
          denyLabel="deny"
        >
          {() => (1)}
        </Modal>, container,
      )
    })
    expect(container).toMatchSnapshot()
  })
})
