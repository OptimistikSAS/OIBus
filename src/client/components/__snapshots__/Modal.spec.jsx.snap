// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal check Modal 1`] = `
<div>
  1
</div>
`;
